### ISSUE TITLE
Modal dialog popup with Exception of type 'Microsoft.FSharp.Compiler.ErrorLogger+UnresolvedPathReferenceNoRange' was thrown.

### ISSUE THREAD

**Issue by abelbraaksma** [Microsoft/visualfsharp#1641]
At some point during editing in VS2015, I received a popup with the following error:

> Exception of type 'Microsoft.FSharp.Compiler.ErrorLogger+UnresolvedPathReferenceNoRange' was thrown.

### Repro steps
I don't (yet) know how to reproduce this. It happened in a medium-sized project with ~150 F# source files, roughly 30k non-comment source lines, itself opened in a solution with 12 projects (most of them smaller). Apart from the relative size of the projects, there's nothing out of the ordinary there (all pure F#).

It happened right after a fast Ctrl-C/Ctrl-V/Ctrl-S cycle (I often hit Ctrl-S because Visual Studio permanently freezes / crashes / disappears so often, esp. during fast edits and ctrl-tab switches). I suspected a faulty file or directory but a quick chkdsk didn't show up anything unusual. The file was open by only one solution and one instance of VS.

After the error appeared, I could save the file normally and continue working with Visual Studio.

### Actual behavior

Here's copied textual representation of the full error:

> \---------------------------
> Microsoft Visual Studio
> \---------------------------
> Exception of type 'Microsoft.FSharp.Compiler.ErrorLogger+UnresolvedPathReferenceNoRange' was thrown.
> \---------------------------
> OK
> \---------------------------

And the screenshot:

![error](http://i.imgur.com/iETSYbw.png)

### Related information 

Using Visual Studio 2015, Update 3, system up-to-date with most recent windows-update updates.

I know I shouldn't post when I don't have a repro, but maybe someone else encountered this and knows more about it, or on where to start further research. So far it happened only once and (apart from the daily VS crashes) the system seems stable.

Update: a similar error [was mentioned here](https://github.com/tpetricek/FSharp.Formatting/issues/218), but seems unrelated, and [on FAKE](https://github.com/fsharp/FAKE/issues/719). Both talk about `libDirs`, but I fail to see how that relates to the VS build environment.

**Comment by abelbraaksma** [comment]
Update: this message box has popped up again since. The Ctrl-C or Ctrl-S related nature I suspected does not hold. It is also thrown when I'm just editing, without doing something fancy like hitting Ctrl-V or saving a file. 

The only common denominator appears to be that the project is not compilable when this exception is thrown. I'd wonder how I could debug it or find the reasons that may cause it to pin it down.

**Comment by dsyme** [comment]
@abelbraaksma Did you have VF# power tools installed?

**Comment by abelbraaksma** [comment]
@dsyme, yes. Working without them is virtually impossible, those tools are a lifesaver (I'd actually hope some of them become part of Visual F# one day). Are you saying that I should try locate this error in the powertools? The nested classname suggests to me it is part of F# itself, no?

**Comment by dsyme** [comment]
@abelbraaksma I'm not sure.  The FSharp.Compiler.Service class names are the same, so it could be from either.  I think without a repro we should close this for now.

**Comment by dsyme** [comment]
@a

**Comment by abelbraaksma** [comment]
@dsyme, I understand. I am willing to look into this deeper, but I don't know how to debug this. If it was a "normal" crash I could attach a debugger, but this is a popup and I have no idea how to set a breakpoint prior to that happening.

It's probable that it is obscure, but not necessarily so: after you dismiss the popup everything seems normal. It wouldn't be the first bug in an application that has bugged people without them raising an issue about it.

If I can repro, I will report so, of course.

**Comment by dsyme** [comment]
@abelbraaksma The main thing would be to catch a stack trace at the point where it occurs

**Comment by abelbraaksma** [comment]
@dsyme: yes, that's the point. There's no exception, there's a popup, so I cannot hook into it when it happens (and it doesn't happen often enough to attach the debugger all the time). I'll browse the sources if I can deduct where this is thrown.

**Comment by abelbraaksma** [comment]
I has since happened at least 7 times (perhaps more, I started counting only a few weeks back), with F# and VS2015 (often multiple instances, sometimes with the same projects in multiple instances, never the same sln files though). 

It has happened on both my laptop (Win10) and my desktop (Win7) system. I would assume that it also happens to others, but that's just me guessing. I haven't ruled out the possibility it being related to a plugin as it is impossible to work long enough on a non-plugged-in system (with the exception of NCrunch/PostSharp/.NET Reflector/Resharper, these I had disabled and it still happened).

It seems to happen when I do something _fast_, esp. with key-combinations involving auto-completions. It does _not_ seem to happen specifically when saving a document, I've been able to assert that much.

So far I've not been able to attach a debugger on, near or at the event. Still unsure where to look. Also still unsure how "unstable" this makes VS or what the error relates to.

Maybe there's a way to find out what parts can cause `UnresolvedPathReferenceNoRange` exceptions and add a `Debugger.Break()` statement and use that build.
